README.md

```diff
@@ -289,6 +301,214 @@ Gulp.js is used for building CSS assets.
 
 In order to build all CSS assets **run `node_modules/gulp/bin/gulp.js` once**. Gulp.js is watching for changes in the source LESS assets and will rebuild CSS on each change. If you have gulp.js installed globally on the system, you can just type `gulp` to run it.
 
+## Custom Development Within the ckanext-querytool Extension
+
+This extension diverges from CKAN core in a number of ways. The following sections cover the major areas of divergence and areas most commonly customized, along with high-level descriptions and links to the relevant code and external documentation.
+
+The frontend libraries and frameworks used by this extension for charts and tables are:
+
+- [Plotly](https://plotly.com/javascript/) for charts (bar, line, pie, etc.)
+- [DataTables](https://datatables.net/) for tables
+- [Leaflet](https://leafletjs.com/) for maps
+
+These are used on top of:
+
+- [jQuery](https://jquery.com/)
+- [Bootstrap](https://getbootstrap.com/)
+- [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
+
+If any changes are made to the JavaScript files, you must be rebuild using [webpack](https://webpack.js.org/) (see the [Updating source files](#updating-source-files) section above).
+
+For more information on CKAN core architecture as a whole, see [CKAN code architecture](https://docs.ckan.org/en/2.7/contributing/architecture.html?highlight=helpers). For anything not covered here, see the [CKAN 2.7 core documentation](https://docs.ckan.org/en/2.7/).
+
+### Frontend
+
+#### Plotly
+
+This extension uses [plotly.js](https://plotly.com/javascript/) for charts (bar, line, pie, etc.). It's a JavaScript library for creating interactive charts and graphs, built on top of [d3.js](https://d3js.org/) and [stack.gl](https://github.com/stackgl).
+
+Most changes and customizations of plotly.js are done in the `ckanext-querytool/ckanext/querytool/fanstatic/javascript/modules/viz-preview.js` file.
+
+The main settings object can be found on [line 1977 of the `viz-preview.js` file](https://github.com/datopian/ckanext-querytool/blob/master/ckanext/querytool/fanstatic/javascript/modules/viz-preview.js#L1977), and it's called `base_info`. It contains the following properties:
```

Comment:
I'm not sure if it's a good idea to point to lines on files, `viz-preview.js` might be updated often (imo), but since you also pasted the code below I think it's not a big issue